### DISCUSSION TITLE
[FEATURE] frequency manager taking bookmark with right click

### DISCUSSION THREAD

**Discussion by Zaryob** [AlexandreRouma/SDRPlusPlus#1475] [Ideas]
While I’m using SdrPP I ‘am taking so many bookmarks. Is there possible to add this feature.

I’m planning to add (and trying) right click event on `Waterfall::processInputs` it emits right click handler and it handled by frequency manager’s module. Is that true approach for SdrPP’s structure?

**Comment by Zaryob** [comment]
Main issue that I’m dealing is for each time ‘menuHandler’ Popup is shown. Otherwise it is not possible to showing until frequency manager uncollapsed. But it works as I need

**Reply by arun1969** [reply]
you can add following function in menu.cpp
	bool Menu::isOpen(std::string name)
	{
		for(MenuOption_t opt : order)
		{
			if(opt.name == name)
				return(opt.open);
		}

		return(false);
	}
and check in add bookmark in frequency manager like.
		if(!gui::menu.isOpen(_this->name))
			return;
